### PR TITLE
Removed deprecated node "viewdirection"

### PR DESCRIPTION
**Pull request by niklasharrysson**
Removing the viewdirection node from stdlib, as discussed we don't want view dependency introduced in our graphs in such a general form.

Since viewdirection is needed for the environment shader in MaterialXView I have made a custom node for this, which is only available for the viewer. This was the least intrusive fix I could find for now. When we have more time we can clean this up and create this environment shader without using a nodegraph.

### CODE REVIEW

File: resources/Lights/envmap_shader.mtlx [hunk 1]
```diff
@@ -1,11 +1,17 @@
 <?xml version="1.0"?>
 <materialx version="1.37">
+
+   <!-- Custom node: <viewdir>                     -->
+   <!-- Required by environment shader graph below -->
+   <nodedef name="ND_viewdir_vector3" node="viewdir">
+     <output name="out" type="vector3" default="0.0, 0.0, 1.0"/>
+   </nodedef>
+  <implementation name="IM_viewdir_vector3_genglsl" nodedef="ND_viewdir_vector3" target="genglsl"/>
```
Comment (niklasharrysson) [inline resources/Lights/envmap_shader.mtlx:9]: This file is loaded explicitly by the viewer, so the <viewdir> node is only available here.

File: source/MaterialXView/Material.cpp [hunk 2]
```diff
@@ -34,8 +34,13 @@ bool Material::loadSource(const mx::FilePath& vertexShaderFile, const mx::FilePa
...
     }
 
+    // TODO:
+    // Here we set new source code on the _glProgram without rebuilding 
+    // the _hwShader instance. So the _hwShader is not in sync with the
+    // _glProgram after this operation.
     _glProgram = mx::GlslProgram::create();
-    _glProgram->setStages(_hwShader);
+    _glProgram->addStage(mx::Stage::VERTEX, vertexShader);
+    _glProgram->addStage(mx::Stage::PIXEL, pixelShader);
```
Comment (niklasharrysson) [inline source/MaterialXView/Material.cpp:43]: Fixed a regression where shader source were no longer loaded here.

File: source/MaterialXView/Viewer.cpp [hunk 3]
```diff
@@ -124,6 +124,52 @@ void applyModifiers(mx::DocumentPtr doc, const DocumentModifiers& modifiers)
     }
 }
 
+// ViewDir implementation for GLSL
+// as needed for the environment shader.
+class ViewDirGlsl : public mx::GlslImplementation
```
Comment (niklasharrysson) [inline source/MaterialXView/Viewer.cpp:129]: Based on the old viewdirection implementation. Now moved here so it's only available for the viewer.

File: source/MaterialXView/Material.cpp (hunk 2 above)
Comment (bernardkwok) [reply source/MaterialXView/Material.cpp:43]: Good to have ths fixed. Thanks.

**Review by bernardkwok — APPROVED** [review]

**Pull request merged**